api/src/main/java/org/openmrs/module/fgh/mpi/MpiHttpClient.java

```diff
@@ -62,32 +47,42 @@ public class MpiHttpClient {
 	 * @throws Exception
 	 */
 	public Map<String, Object> getPatient(String patientUuid) throws Exception {
+		MpiContext mpiContext = MpiContext.initIfNecessary();
+		
 		log.info("Looking up patient record from MPI with OpenMRS uuid: " + patientUuid);
 		
 		if (log.isDebugEnabled()) {
 			log.debug("Searching for patient from MPI with OpenMRS uuid: " + patientUuid);
 		}
 		
-		if (openmrsUuidSystem == null) {
-			synchronized (MpiHttpClient.class) {
-				openmrsUuidSystem = MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM);
-			}
-		}
+		String query = REQ_PARAM_SOURCE_ID + "=" + mpiContext.getOpenmrsUuidSystem() + "|" + patientUuid;
+		
+		Map<String, Object> pixResponse = null;
+		
+		pixResponse = submitRequest(SUBPATH_PATIENT + "/$ihe-pix?" + query, null, Map.class);
 		
-		String query = REQ_PARAM_SOURCE_ID + "=" + openmrsUuidSystem + "|" + patientUuid;
-		Map<String, Object> pixResponse = submitRequest(SUBPATH_PATIENT + "/$ihe-pix?" + query, null, Map.class);
 		List<Map<String, Object>> ids = (List<Map<String, Object>>) pixResponse.get(RESPONSE_FIELD_PARAM);
-		if (ids.isEmpty()) {
+		if (ids == null || ids.isEmpty()) {
```

Comment:
You can replace this with a single call of MapUtils.isEmpy(ids)

Reply:
This is a list of Map and is not applicable for isEmpty method.

Reply:
I mean you can use CollectionsUtils

Reply:
Done